### PR TITLE
added leverj file in bin directory and market_maker_keeper

### PR DESCRIPTION
**Pull request by mitakash**

### CODE REVIEW

File: bin/leverj-market-maker-keeper [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+dir="$(dirname "$0")"/..
+source $dir/_virtualenv/bin/activate || exit
+export PYTHONPATH=$PYTHONPATH:$dir:$dir/lib/pymaker:$dir/lib/pyexchange:$dir/lib/ethgasstation-client:$dir/lib/gdax-client
```
Comment (EdNoepel) [inline bin/leverj-market-maker-keeper]: Why does leverj need `gdax-client` in `PYTHONPATH`?

File: market_maker_keeper/leverj_market_maker_keeper.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
+# This file is part of Maker Keeper Framework.
+#
+# Copyright (C) 2018 ith-harvey, grandizzy
```
Comment (EdNoepel) [inline market_maker_keeper/leverj_market_maker_keeper.py]: Copyright (C) 2019 mitakash

File: market_maker_keeper/leverj_market_maker_keeper.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    def token_buy(self) -> str:
+        if self.arguments.pair.startswith("USDC"):
+            return self.arguments.pair[4:]
+        return self.arguments.pair[3:]
+
+
+    def our_available_balance(self, our_balances: dict, token: str) -> Wad:
+        for key in our_balances:
+            if our_balances[key]['symbol'] == token:
+                if (token == "LEV") or (token == "FEE"):
```
Comment (EdNoepel) [inline market_maker_keeper/leverj_market_maker_keeper.py]: Does the exchange provide metadata describing the number of decimal places for each token?  Seems dangerous to hardcode this.

File: market_maker_keeper/leverj_market_maker_keeper.py [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+                if (token == "LEV") or (token == "FEE"): 
+                    return Wad(int(our_balances[key]['available'])*10**9)
+                elif (token == "USDC") or (token == "USDT"):
+                    return Wad(int(our_balances[key]['available'])*10**12)
+                elif (token == "WBTC"):
+                    return Wad(int(our_balances[key]['available'])*10**10)
+                elif (token == "GUSD"):
+                    return Wad(int(our_balances[key]['available'])*10**16)
+                else:
+                    return Wad(int(our_balances[key]['available']))
```
Comment (EdNoepel) [inline market_maker_keeper/leverj_market_maker_keeper.py]: Wouldn't the default case require multiplying by `10**18`, or using `Wad.from_number(...)`?

File: bin/leverj-market-maker-keeper (hunk 1 above)
Comment (mitakash) [reply bin/leverj-market-maker-keeper]: basically for price we eth-dai for live quoting which it gets from gdax

File: market_maker_keeper/leverj_market_maker_keeper.py (hunk 3 above)
Comment (mitakash) [reply market_maker_keeper/leverj_market_maker_keeper.py]: No, I think it's dangerous to rely on exchange meta data for this. If the exchange screws up this very basic information, we will be the ones that lose money on their error. It's better to hard code this or get this data from our side. On that note , if model.py was approved in pymaker, I could probably use that and our configs to pass in this data.

File: market_maker_keeper/leverj_market_maker_keeper.py (hunk 4 above)
Comment (mitakash) [reply market_maker_keeper/leverj_market_maker_keeper.py]: No, for normal tokens the exchange passes back strings that in the base number, so for normal tokens like DAI, it's already in 10**18 format but just in a string case.

File: market_maker_keeper/leverj_market_maker_keeper.py [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+        self.order_book_manager.start()
+
+    def main(self):
+        with Lifecycle() as lifecycle:
+            lifecycle.initial_delay(1)
+            lifecycle.on_startup(self.startup)
+            lifecycle.every(1, self.synchronize_orders)
+            lifecycle.on_shutdown(self.shutdown)
+
+    def startup(self):
```
Comment (ith-harvey) [inline market_maker_keeper/leverj_market_maker_keeper.py]: can you explain the difference between the precision here and the precision you get / use in pyexchange?

File: market_maker_keeper/leverj_market_maker_keeper.py [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+        new_orders = bands.new_orders(our_buy_orders=self.our_buy_orders(order_book.orders),
+                                      our_sell_orders=self.our_sell_orders(order_book.orders),
+                                      our_buy_balance=self.our_available_balance(order_book.balances, self.token_buy()),
+                                      our_sell_balance=self.our_available_balance(order_book.balances, self.token_sell()),
+                                      target_price=target_price)[0]
+        self.place_orders(new_orders)
+
+    def place_orders(self, new_orders: List[NewOrder]):
+        def place_order_function(new_order_to_be_placed):
+            price = round(new_order_to_be_placed.price, self.precision + 2)
```
[the remark this answers is not in the document]
Comment (mitakash) [reply market_maker_keeper/leverj_market_maker_keeper.py]: so we have extra precision in market maker keeper to do some calculations, but finally, when we send the order in pyexchange we limit it to the precision of the exchange.

File: market_maker_keeper/leverj_market_maker_keeper.py (hunk 5 above)
Comment (mitakash) [reply market_maker_keeper/leverj_market_maker_keeper.py]: so we have extra precision in market maker keeper to do some calculations, but finally, when we send the order in pyexchange we limit it to the precision of the exchange.

File: market_maker_keeper/leverj_market_maker_keeper.py (hunk 3 above)
Comment (mitakash) [reply market_maker_keeper/leverj_market_maker_keeper.py]: Let's go with hardcoded, this is info that you can easily screw up with configs, and the consequences are large if you mess up in the config

**Pull request merged**